slangpy/testing/benchmark/fixtures.py

```diff
@@ -4,20 +4,62 @@
 import slangpy as spy
 from slangpy.core.function import FunctionNodeBwds
 import numpy as np
-from typing import Any, Union
+from typing import Any, Callable, Union
 from time import time
 from datetime import datetime
 
 from .report import BenchmarkReport
 
 
-class BenchmarkFixture:
+class ReportFixture:
 
     def __init__(self, config: pytest.Config, node: Any):
         super().__init__()
         self.config = config
         self.node = node
 
+    def __call__(
+        self,
+        device: spy.Device,
```

Comment:
Should the report fixture not take a `device` in but an additional `meta` parameter?